CustomContent: Add option to fetch only topics

manuq opened the pull request:
Add a onlyTopics option to the fetchContentCollection method. And add
the same option in the ContentNodeFilter of the kolibri API.

https://phabricator.endlessm.com/T32561

<!--
 1. Following guidance below, replace …'s with your own words
 2. After saving the PR, tick of completed checklist items
 3. Skip checklist items that are not applicable or not necessary
 4. Delete instruction/comment blocks
-->

### Summary
<!--
 * description of the change
 * manual verification steps performed
 * screenshots if the PR affects the UI
-->

…

### Reviewer guidance
<!--
 * how can a reviewer test these changes?
 * are there any risky areas that deserve extra testing
-->

…

### References
<!--
 * references to related issues and PRs
 * links to mockups or specs for new features
 * links to the diffs for any dependency updates, e.g. in iceqube or the perseus plugin
-->

…

----

### Contributor Checklist


PR process:

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

manuq committed:
CustomContent: Add option to fetch only topics

Add a onlyTopics option to the fetchContentCollection method. This is
a shortcut to pass `kind: topic` to the kolibri API.

https://phabricator.endlessm.com/T32561

danigm commented on kolibri/core/content/api.py:
```diff
@@ -160,6 +160,7 @@ class ContentNodeFilter(IdFilter):
     # Use a CharFilter instead a BooleanFilter to be able to avoid cache using
     # different values
     random = CharFilter(method="filter_random")
+    only_topics = BooleanFilter(method="filter_only_topics")
```
I think this filter is not needed, it does the same as the `kind` filter.

manuq replied:
Yes, this makes me realize that we should do the same for `onlyContent`. In the backend, when `onlyContent` is passed, we just use the `kind_in`. Then there is no need to change the `api.py` in the Python side.

danigm commented on kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue:
```diff
@@ -120,6 +120,7 @@
             channel_id: this.topic.id,
             page: options.page ? options.page : 1,
             page_size: options.pageSize ? options.pageSize : 50,
+            only_topics: options.onlyTopics ? true : undefined,
```
We can use the `kind` filter here, as I said above:

```suggestion
            kind: options.onlyTopics ? 'topic' : undefined,
```

manuq committed:
CustomContent: Make onlyContent a shortcut for kind_in

And remove the filter previously added in the kolibri api.

https://phabricator.endlessm.com/T32561

danigm commented on kolibri/plugins/learn/assets/src/views/ChannelRenderer/CustomContentRenderer.vue:
```diff
@@ -29,11 +29,16 @@
 
+  const allButTopicTypes = Object.keys(ContentNodeKinds).filter(
+    k => k === ContentNodeKinds.TOPIC
+  );
```
I think this won't work, the API expect to receive a [comma separated list](https://github.com/endlessm/kolibri/blob/release-v0.14.x/kolibri/core/content/api.py#L192-L201) of content kind.

And we should also use the `Object.values`, because the keys are in uppercase and didn't match the kind in all cases.

This should b e something like:

```
  const allButTopicTypes = Object.values(ContentNodeKinds).filter(k => k === 'topic').join(',');
```

manuq replied:
You are right about the values! And the filter should be for `!==` instead of `===`. 

There is no need for converting to a string of comma separated values, because the serialization takes care of that already.
